### DISCUSSION TITLE
Importing Operational Capabilities - Issue with Names

### DISCUSSION THREAD

**Discussion by merlyn-true** [labs4capella/python4capella#206] [Q&A] [answered]
Hello, 
I am trying to write a script that imports Operational Capabilities from an Excel spreadsheet.

I modified one of the sample scripts provided for this purpose. 

The script works for importing the correct number of Capabilities, however it does not name them according to what is listed. 

The default name is kept for some reason.

I suspect the lines giving the issue are:
            oc_name = cell.value
            oc.set_name(oc_name)

So far I've tried pulling out the cell.value into a variable (shown above).
I've also tried oc.set_name(str(cell.value)). This also did not work. 

Wondering if anyone has encountered this or could provide me with some insights. 

Thank you

Here is my code
`'''
Created on Jan. 27, 2024

@author: Patrice Balthazar
'''

include('workspace://Python4Capella/simplified_api/capella.py')
if False:
    from simplified_api.capella import *
    
# include needed for utilities
include('workspace://Python4Capella/utilities/CapellaPlatform.py')
if False:
    from utilities.CapellaPlatform import *

# include needed to read/write xlsx files
from openpyxl import *

#check parameter numbers
if len(argv) != 2:
    # use IFE default values
    aird_path = "/test-project/test-project.aird"
    xlsx_path = "/scripting-testing/excel-files/test_op_capabilities_import.xlsx"
   # xlsx_path = "/Python4Capella/resources/physical_components.xlsx"
else:
    # Load the Capella model from the first argument of the script
    aird_path = argv[0]
    xlsx_path = argv[1]
 
model = CapellaModel()
model.open(aird_path)

# # gets the SystemEngineering
se = model.get_system_engineering()

# create  a folder in the project
xlsx_file = CapellaPlatform.getWorkspaceFile(xlsx_path)
# print(xlsx_file)
xlsx_file_name = CapellaPlatform.getAbsolutePath(xlsx_file)
#
print("Read " + xlsx_file_name)

# load the workbook
wb = load_workbook(xlsx_file_name)

# grab the active worksheet
ws = wb.active

# get the OperationalCapabilityPkg
oc_pkg = se.get_operational_analysis().get_operational_capability_pkg()

# start a transaction to modify the Capella model
model.start_transaction()
try:
    # create operational capabilities with the list of capabilities in the xlsx file
    for row in ws.iter_rows():
        for cell in row:
            # create an OperationalCapability
            oc = OperationalCapability()
            #set its name
            print("* Create a new operational capability with name " + cell.value)
            oc_name = cell.value
            oc.set_name(oc_name)
            #add the new OperationalCapability
            oc_pkg.get_owned_operational_capabilities().add(oc)
            org.polarsys.capella.core.model.helpers.CapellaElementExt.creationService(oc.get_java_object())
except:
    # if something went wrong we rollback the transaction
    model.rollback_transaction()
    raise
else:
    # if everything is ok we commit the transaction
    model.commit_transaction()

# save the Capella model
model.save()

`

**Comment by ylussaud** [comment, marked as answer]
```
org.polarsys.capella.core.model.helpers.CapellaElementExt.creationService(oc.get_java_object())
```

This method has the size effect of renaming the passed element with a default name. You need to set the object features after calling this method:

```
org.polarsys.capella.core.model.helpers.CapellaElementExt.creationService(oc.get_java_object())
oc.set_name(oc_name)
oc....
```

**Reply by merlyn-true** [reply]
Thank you for the prompt response.
I made the modification and it worked.


I originally did make this change because the sample script for adding physical components from xlsx is written in that order (pc name is set prior to the line).

I think this should be changed in the sample script as well :) [if not done yet].

Thanks again.

**Reply by ylussaud** [reply]
Yes I had the same issue :)
It's fixed now in the sample script.

BTW: There is a [forum for Python4Capella](https://forum.mbse-capella.org/c/scripting-with-capella-python4capella/23) where you can find already answered questions and also ask your own.
